File: backend/app/schemas/publish_task.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, List, Any
from datetime import datetime


class PublishTaskBase(BaseModel):
    content_title: str = Field(..., description="内容标题")
    content_description: Optional[str] = None
    platform_account_id: int
    platform: str = Field(..., description="平台类型")
    scheduled_at: Optional[datetime] = None
    tags: Optional[List[str]] = None
    category: Optional[str] = None
    video_url: Optional[str] = None
    cover_url: Optional[str] = None


class PublishTaskCreate(PublishTaskBase):
    video_project_id: Optional[int] = None


class PublishTaskUpdate(BaseModel):
    content_title: Optional[str] = None
    content_description: Optional[str] = None
    scheduled_at: Optional[datetime] = None
    tags: Optional[List[str]] = None
    status: Optional[str] = None
    review_status: Optional[str] = None
    review_comment: Optional[str] = None


class PublishTaskMetrics(BaseModel):
    play_count: int = 0
    like_count: int = 0
    comment_count: int = 0
    share_count: int = 0
    collect_count: int = 0
    follower_gained: int = 0
    completion_rate: Optional[float] = None
    ctr: Optional[float] = None


class PublishTaskResponse(PublishTaskBase):
    id: int
    user_id: int
    video_project_id: Optional[int]
    status: str
    published_at: Optional[datetime]
    platform_post_id: Optional[str]
    platform_post_url: Optional[str]
    publish_error: Optional[str]
    metrics_collected_at: Optional[datetime]
    play_count: int
    like_count: int
    comment_count: int
    share_count: int
    collect_count: int
    follower_gained: int
    completion_rate: Optional[float]
    ctr: Optional[float]
    growth_insight: Optional[Dict[str, Any]]
    processed_by_graph: int
    review_status: Optional[str]
    review_comment: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class PublishTaskListResponse(BaseModel):
    items: List[PublishTaskResponse]
    total: int

File: frontend/src/api/publishTask.ts
import request from '@/utils/request'

export function getPublishTasks(params?: { status?: string; platform?: string }) {
  return request.get('/publish-tasks', { params })
}

export function createPublishTask(data: {
  content_title: string
  content_description?: string
  platform_account_id: number
  platform: string
  video_project_id?: number
  scheduled_at?: string
  tags?: string[]
  category?: string
  video_url?: string
  cover_url?: string
}) {
  return request.post('/publish-tasks', data)
}

export function updatePublishTask(id: number, data: Partial<{
  content_title: string
  content_description: string
  scheduled_at: string
  tags: string[]
  status: string
  review_status: string
  review_comment: string
}>) {
  return request.put(`/publish-tasks/${id}`, data)
}

export function deletePublishTask(id: number) {
  return request.delete(`/publish-tasks/${id}`)
}

export function triggerPublish(id: number) {
  return request.post(`/publish-tasks/${id}/publish`)
}

export function collectMetrics(id: number, data: {
  play_count?: number
  like_count?: number
  comment_count?: number
  share_count?: number
  collect_count?: number
  follower_gained?: number
  completion_rate?: number
  ctr?: number
}) {
  return request.post(`/publish-tasks/${id}/collect-metrics`, data)
}

export function autoCollectMetrics(id: number) {
  return request.post(`/publish-tasks/${id}/auto-collect`)
}

export function autoReviewTask(id: number) {
  return request.post(`/publish-tasks/${id}/auto-review`)
}

export function runFullPipeline(id: number) {
  return request.post(`/publish-tasks/${id}/full-pipeline`)
}
